### PR TITLE
test case and proposed fix for issue #182, sub sockets not reconnecting

### PR DESCRIPTION
**Pull request by TTimo**
test case and proposed fix for issue #182, sub sockets not reconnecting
refreshed and tested against latest master

### CODE REVIEW

**Comment by djc** [comment]
Also, it would be nice if you can rewrite the commit message to have a moree concise first line (e.g. max 70 characters or so) and more explanation in subsequent lines.

File: src/core/sock.c [hunk 1]
```diff
@@ -658,6 +658,12 @@ int nn_sock_recv (struct nn_sock *self, struct nn_msg *msg, int flags)
             return rc;
         }
 
+		/* Have to check if one broken connection needs a reconnection at all cases */
+		if(nn_fast (rc == -EAGAIN) && nn_fast(self->statistics.broken_connections > 0)) {
```
Comment (gdamore) [inline src/core/sock.c]: nn_fast feels wrong here.  For two reasons:

1. We're already in an error path -- this should never be hot -- feels like premature optimization
2. Is it the case that when in this error path, this is really the most  likely case?

There are indent and spacing issues as Dirkjan already noted, as well.

File: tests/ipc_pub_disconnect.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/*
+    Copyright (c) 2012 250bpm s.r.o.  All rights reserved.
```
Comment (gdamore) [inline tests/ipc_pub_disconnect.c]: This copyright is wrong.

**Comment by gdamore** [comment]
As I indicated in #182 the approach here is wrong -- we need to rethink this and solve it without kicking the error back to the application.  Maybe hard to do without bringing in threads...

**Pull request closed without merging**